ext/openssl/openssl_arginfo.h

```diff
@@ -536,45 +536,162 @@ static const zend_function_entry class_OpenSSLAsymmetricKey_methods[] = {
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_pkey_export_to_file", 2);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_pkey_export", 0);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_pkey_export", 2);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_pkey_get_private", 0);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_pkey_get_private", 1);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_get_privatekey", 0);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_get_privatekey", 1);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_pbkdf2", 0);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_pkcs7_sign", 3);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_pkcs7_decrypt", 2);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_pkcs7_decrypt", 3);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_cms_sign", 3);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_cms_decrypt", 2);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_cms_decrypt", 3);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_private_encrypt", 0);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_private_encrypt", 2);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_private_decrypt", 1);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_private_decrypt", 2);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_public_encrypt", 0);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_public_decrypt", 1);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_sign", 2);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_seal", 0);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_open", 1);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_open", 3);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_encrypt", 0);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_encrypt", 2);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_decrypt", 2);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_dh_compute_key", 1);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_pkey_derive", 1);
-	zend_mark_function_parameter_as_sensitive(CG(function_table), "openssl_spki_new", 0);
+
+	zend_string *attribute_name_SensitiveParameter_arginfo_openssl_x509_check_private_key_arg1 = zend_string_init("SensitiveParameter", sizeof("SensitiveParameter") - 1, 1);
```

Comment:
I'm wondering if it would be beneficial to declare the different attribute names first and then use them in the `zend_add_parameter_attribute()` invocations. It would save quite a few lines when an extension has a lot of attributes with the same name

Reply:
It likely would, but that should probably be a follow-up, as it also affects the class attributes.